test/unit/gen/lister.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2019 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gen
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	clientcorev1 "k8s.io/client-go/listers/core/v1"
+
+	"github.com/jetstack/cert-manager/pkg/controller/test/fake"
+)
+
+type FakeSecretListerModifier func(*fake.FakeSecretLister)
+type FakeSecretNamespaceListerModifier func(*fake.FakeSecretNamespaceLister)
+
+func FakeSecretLister(mods ...FakeSecretListerModifier) *fake.FakeSecretLister {
```

Comment:
Is this necessary? If so, I think it'd be better to move it out of this package.. whilst maybe poorly named, `test/unit/gen` is meant to be for generating API types and this pulls in a dependency on `client-go` to this package 😅